runner_manager/models/runner.py

```diff
@@ -151,16 +152,23 @@ def time_to_start_expired(self, timeout: timedelta) -> bool:
     def time_to_live_expired(self, time_to_live: timedelta) -> bool:
         return self.is_active and self.time_since_started > time_to_live
 
-    def update_from_github(self, github: GitHub) -> "Runner":
+    def update_from_github(
+        self, github: GitHub, headers: Optional[Dict[str, str]] = None
+    ) -> "Runner":
         if self.id is not None:
-            github_runner: GitHubRunner = (
-                github.rest.actions.get_self_hosted_runner_for_org(
-                    org=self.organization, runner_id=self.id
-                ).parsed_data
-            )
-            self.status = RunnerStatus(github_runner.status)
-            self.busy = github_runner.busy
-            log.info(f"Runner {self.name} status updated to {self.status}")
+            try:
+                github_runner: GitHubRunner = (
+                    github.rest.actions.get_self_hosted_runner_for_org(
+                        org=self.organization, runner_id=self.id, headers=headers
+                    ).parsed_data
+                )
+                self.status = RunnerStatus(github_runner.status)
+                self.busy = github_runner.busy
+            except RequestFailed:
+                log.info(f"Runner {self.name} does not exist anymore.")
+                self.status = RunnerStatus.offline
+                self.busy = False
```

Comment:
As we use a specific Exception, I guess it can happen only in the first call of the try statement, so I would prefer having this instead:

```suggestion
            except RequestFailed:
                log.info(f"Runner {self.name} does not exist anymore.")
                self.status = RunnerStatus.offline
                self.busy = False
            else:
                self.status = RunnerStatus(github_runner.status)
                self.busy = github_runner.busy
```

That's more pythonic.

Reply:
Didn't knew we could do that, I like it.